### DISCUSSION TITLE
Why need to restart gateway?

### DISCUSSION THREAD

**Discussion by 9to1url** [narrowlink/narrowlink#25] [General]
I am not sure this is a bug so just bring up in Discussions.

I found I can't connect after 2 days not use, getting this error:
```
[2023-08-20T15:09:03Z WARN  narrowlink] Unable to connect server: E-UnableToUpgrade, Error(403)
```

So I restarted the Agent and Client, but still not working.
So I restarted Gateway and Client again, that starts to work.

I am not sure why I need to restart gateway? Any idea?  :-)

**Comment by SajjadPourali** [comment]
Most likely, this is a bug. The gateway does not need it at all. There are a few mishandled errors on the gateway side that I have plans to address at https://github.com/narrowlink/narrowlink/issues/21. However, the likelihood of encountering these errors is very low; they may only occur on extremely unstable networks.

In any case, I have given the highest priority to addressing such issues.

I'm going to check the code more carefully. If you set this environment variable on both the client and gateway sides, you can get more logs that would be very helpful for me.

```bash
RUST_LOG=narrowlink,narrowlink_gateway,narrowlink_types,narrowlink_network=trace
```
like:
```bash
export RUST_LOG=narrowlink,narrowlink_gateway,narrowlink_types,narrowlink_network=trace
narrowlink-gateway > logs.txt
```
or
```bash
RUST_LOG=narrowlink,narrowlink_gateway,narrowlink_types,narrowlink_network=trace ./narrowlink-gateway > logs.txt
```
